src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -57,62 +61,106 @@ private HttpClient httpClient
         /// <summary>
         /// Initializes a new instance of the <see cref="ChangeAnalysisClient"/> class.
         /// </summary>
-        public ChangeAnalysisClient(ChangeAnalysisDataProviderConfiguration configuration, string clientObjectId, string clientPrincipalName = "")
+        public ChangeAnalysisClient(ChangeAnalysisDataProviderConfiguration configuration, string requestTrackingId, string clientObjectId, string clientPrincipalName = "")
         {
             clientObjectIdHeader = clientObjectId;
             clientPrincipalNameHeader = clientPrincipalName;
             changeAnalysisEndPoint = configuration.Endpoint;
             apiVersion = configuration.Apiversion;
+            requestId = requestTrackingId;
         }
 
         /// <inheritdoc/>
         public async Task<List<ResourceChangesResponseModel>> GetChangesAsync(ChangeRequest changeRequest)
         {
-            string requestUri = changeAnalysisEndPoint + $"changes?api-version={apiVersion}";
-            object postBody = new
-            {
-                changeRequest.ResourceId,
-                changeRequest.ChangeSetId
-            };
-            string jsonString = await PrepareAndSendRequest(requestUri, postBody, HttpMethod.Post);
-            List<ResourceChangesResponseModel> resourceChangesResponse = JsonConvert.DeserializeObject<List<ResourceChangesResponseModel>>(jsonString);
-            return resourceChangesResponse;
+            try
+            {
+                string requestUri = changeAnalysisEndPoint + $"changes?api-version={apiVersion}";
+                object postBody = new
+                {
+                    changeRequest.ResourceId,
+                    changeRequest.ChangeSetId
+                };
+                string jsonString = await PrepareAndSendRequest(requestUri, postBody, HttpMethod.Post);
+                List<ResourceChangesResponseModel> resourceChangesResponse = JsonConvert.DeserializeObject<List<ResourceChangesResponseModel>>(jsonString);
+                return resourceChangesResponse;
+            }
+            catch (HttpRequestException httpException)
+            {
+                // Its possible that users dont have access to view the change details, log the exception and send empty list of change details.
+                string message = $"HttpRequestException in GetChangesAsync, Message: {httpException.Message} ";
+                if (httpException.Data.Contains(ExceptionStatusCode))
+                {
+                    message += $"Status Code: {httpException.Data[ExceptionStatusCode]}";
+                }
+
+                DiagnosticsETWProvider.Instance.LogDataProviderMessage(requestId, "ChangeAnalysisClient", message);
```

Comment:
I believe this should be warning instead of message?

Reply:
You mean `DataProviderException` instead of `DataProviderMessage`?

Reply:
We don't have an event with warning event level, perhaps Rekha you can create one.

Reply:
I can change it to log as exception for now. I don't think I will have enough time to make it warning level. I can make that change in a separate PR

Reply:
Ok. File a backlog item to create a warning data provider event and for the time, use info event only. Using exception will not be the right thing there.

Reply:
Ok sure